modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue

```diff
@@ -308,26 +310,32 @@ export default {
       }
       try {
         // Returns the doc as represented in the new locale and mode
-        const modeDoc = await apos.http.post(`${apos.login.action}/set-context`, {
-          body: {
-            mode,
-            locale: apos.locale,
-            _id: doc._id
+        const action = (doc.slug.match(/^\//) ? self.apos.page : self.apos.modules[doc.type]).action;
+        const modeDoc = await apos.http.get(`${action}/${doc._id}`, {
+          qs: {
+            'apos-mode': mode,
+            'apos-locale': locale
           }
         });
+        if (navigate && (!modeDoc._url)) {
+          await apos.alert({
+            heading: 'Page Does Not Exist Yet',
+            description: `The page that provides a listing for this type of piece is not yet available as ${mode} in the ${locale} locale.`
+          });
+          return;
+        }
```

Comment:
this will get axed when we land second chance 404 draft mode, yes? Should we add that comment?

Reply:
This PR contains second chance 404 draft mode, which is working so far as I know. But, this alert is a good fallback because the setContext API can also be called during the time you're on the page, and because of potential situations with locales in the future. The second chance 404 mode is only for one case: (1) you arrived on the page "naturally" via a page transition, and (2) it's about to 404.